src/fidesops/task/graph_task.py

```diff
@@ -410,16 +410,17 @@ def filter_data_categories(
             )
         )
 
-        if not target_fields:
+        if not target_field_paths:
             continue
 
-        for row in results:
-            filtered_access_results[node_address].append(
-                {
-                    field: result
-                    for field, result in row.items()
-                    if field in {target.string_path for target in target_fields}
-                }
-            )
+        # Normalize nested data into a flat dataframe
+        df = pd.json_normalize(results, sep=".")
```

Comment:
ahh dataframes takes me back to my R days. Could we include the `DataFrame` return type for clarity here?

Reply:
helpful thank you